ta4j-core/src/main/java/org/ta4j/core/analysis/criteria/NumberOfPositionsCriterion.java

```diff
@@ -33,18 +35,60 @@
  */
 public class NumberOfPositionsCriterion extends AbstractAnalysisCriterion {
 
+    private final PositionFilter positionFilter;
+
+    /**
+     * Constructor.
+     * 
+     * <p>
+     * For counting the number of all positions.
+     */
+    public NumberOfPositionsCriterion() {
+        this.positionFilter = null;
+    }
+
+    /**
+     * Constructor.
+     * 
+     * @param positionFilter consider either the profit or the loss position
+     */
+    public NumberOfPositionsCriterion(PositionFilter positionFilter) {
+        this.positionFilter = Objects.requireNonNull(positionFilter);
+    }
+
     @Override
     public Num calculate(BarSeries series, Position position) {
+        if (positionFilter == PositionFilter.PROFIT) {
+            return position.hasProfit() ? series.numOf(1) : series.numOf(0);
+        }
+        if (positionFilter == PositionFilter.LOSS) {
+            return position.hasLoss() ? series.numOf(1) : series.numOf(0);
+        }
         return series.numOf(1);
     }
 
     @Override
     public Num calculate(BarSeries series, TradingRecord tradingRecord) {
+        if (positionFilter == PositionFilter.PROFIT) {
+            long numberOfWinningPositions = tradingRecord.getPositions().stream().filter(Position::hasProfit).count();
+            return series.numOf(numberOfWinningPositions);
+        }
+        if (positionFilter == PositionFilter.LOSS) {
+            long numberOfLosingPositions = tradingRecord.getPositions().stream().filter(Position::hasLoss).count();
+            return series.numOf(numberOfLosingPositions);
+        }
         return series.numOf(tradingRecord.getPositionCount());
+
     }
 
     @Override
     public boolean betterThan(Num criterionValue1, Num criterionValue2) {
+        if (positionFilter == PositionFilter.PROFIT) {
+            return criterionValue1.isGreaterThan(criterionValue2);
+        }
+        if (positionFilter == PositionFilter.LOSS) {
+            return criterionValue1.isLessThan(criterionValue2);
+        }
```

Comment:
`betterThan(c1, c2) `returns `true` if c1 is better than c2. So that means the comparison has nothing to do with `this.positionFilter` nor? We have to consider positionFilter of c1 and c2 and maybe throw an exception if positionFilters are not equal

Reply:
> So that means the comparison has nothing to do with this.positionFilter nor

@team172011 I don't understand what you mean. In general, comparision depends on `this.positionFilter`. 

- Line 86-88 origin is https://github.com/ta4j/ta4j/blob/b6bad621a49260364a8450bd02539dd61373963a/ta4j-core/src/main/java/org/ta4j/core/analysis/criteria/NumberOfWinningPositionsCriterion.java#L49
- Line 89-91 origin is https://github.com/ta4j/ta4j/blob/b6bad621a49260364a8450bd02539dd61373963a/ta4j-core/src/main/java/org/ta4j/core/analysis/criteria/NumberOfLosingPositionsCriterion.java#L49
- Line 92 origin is https://github.com/ta4j/ta4j/blob/b6bad621a49260364a8450bd02539dd61373963a/ta4j-core/src/main/java/org/ta4j/core/analysis/criteria/NumberOfPositionsCriterion.java#L48

Or did you mean something else?

Reply:
ah yes, you're right this are `Num`